### ISSUE TITLE
Pytorch implementation for Temperature Scaling

### ISSUE THREAD

**Issue by D4V1ND** [Cortys/SEP-Probly-WS2526#121]
### **Motivation**

This epic aims to introduce **model calibration** as a post-processing technique to improve the reliability of model confidences. Modern neural networks exhibit a strange phenomenon where they output overconfident (poorly calibrated confidences). 

Proper calibration is critical for real-world applications (example: self-driving cars or medical diagnosis) where confidence scores are used for  critical decision-making process. 

We will start by implementing **Temperature Scaling (TS**). TS is a single-parameter variant of Platt Scaling and is identified as the simplest and straightforward calibration method. 

### **Core Goals / Tasks**
1. Implement the Temperature Scaling Function: 
- Develop a calibration function in **PyTorch** that accepts the model's raw output (the logit vector zi) and applies a single scalar parameter T > 0 accross all classes.
- Regerence: [Guo et al., On Calibration of Modern Neural Networks](https://arxiv.org/pdf/1706.04599), page 6
<img width="299" height="53" alt="Image" src="https://github.com/user-attachments/assets/5bb54f8a-756c-495a-9189-54758025ad9e" />

2. Optimize the parameter T: 
- Ensure the optimization process is stable and returns a single scalar value for T. 

3. Verify accuracy preservation (**Critical Check**): 
- Add unit tests verifying that reliability improves after scaling.
